File: spec/javascript/test_expr_io_eof.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('ExprIoEof', 'src/fixed_struct.bin', function(r) {
  assert.equal(r.substream1.one, 1262698832);
  assert.strictEqual(r.substream1.two, undefined);

  assert.equal(r.substream2.one, 4294914349);
  assert.equal(r.substream2.two, 1262698832);
});

File: spec/python/test_expr_io_eof.py
import unittest

from expr_io_eof import ExprIoEof

class TestExprIoEof(unittest.TestCase):
    def test_expr_io_eof(self):
        r = ExprIoEof.from_file("src/fixed_struct.bin")

        self.assertEqual(r.substream1.one, 1262698832)
        self.assertEqual(hasattr(r.substream1, 'two'), False)

        self.assertEqual(r.substream2.one, 4294914349)
        self.assertEqual(r.substream2.two, 1262698832)
